src/components/Tutorial/__tests__/Tutorial.ts

```diff
@@ -0,0 +1,260 @@
+      await user.keyboard('{Enter}')
+      expect(screen.getByText('Now type something. Anything will do.')).toBeInTheDocument()
+    })
+
+    it('we create our first thought', async () => {
+      await user.type(lastEmptyThought(), 'first')
+      await user.keyboard('{Enter}')
+      await act(vi.runOnlyPendingTimersAsync)
+
+      expect(screen.getByText('Well done!')).toBeInTheDocument()
+    })
+  })
+
+  it('we create a second thought', async () => {
+    expect(screen.getByText(/Try adding another thought/)).toBeInTheDocument()
+    await user.type(lastEmptyThought(), 'second')
+    await user.keyboard('{Enter}')
+    await act(vi.runOnlyPendingTimersAsync)
+
+    expect(screen.getByText('Now type some text for the new thought.')).toBeInTheDocument()
+  })
+
+  it('we create a third thought', async () => {
+    await user.type(lastEmptyThought(), 'third')
+    await user.keyboard('{Enter}')
+    await act(vi.runOnlyPendingTimersAsync)
+
+    expect(screen.getByText(/Hit the Delete key to delete the current blank thought/)).toBeInTheDocument()
+  })
+
+  it('we learn to create nested thoughts', async () => {
+    /* thoughts:
+  - first
+  - second
+  - third
+  -
+    */
+    await user.keyboard('{Backspace}')
+    await user.keyboard('{Control>}{Enter}{/Control}')
+    await act(vi.runOnlyPendingTimersAsync)
+
+    await user.type(lastEmptyThought(), 'child of third')
+    await act(vi.runOnlyPendingTimersAsync)
+
+    expect(screen.getByText(/As you can see, the new thought "child of third" is nested/)).toBeInTheDocument()
+  })
+
+  describe('step autoexpand', async () => {
+    it('we learn about automatic thought hiding', async () => {
+      await user.click(screen.getByText('Next'))
+      expect(screen.getByText(/thoughts are automatically hidden when you click away/)).toBeInTheDocument()
+    })
+
+    it('we hide nested thoughts by clicking away', async () => {
+      /* thoughts:
+      - first
+      - second
+      - third
+        - child of third
+      */
+      await user.click(screen.getByText('second'))
+      expect(screen.getByText(/Notice that "child of third" is hidden now/)).toBeInTheDocument()
+    })
+
+    it('we reveal hidden thoughts by clicking parent', async () => {
+      await user.click(screen.getAllByText('third').at(-1)!)
+      expect(screen.getByText('child of third')).toBeInTheDocument()
+    })
+  })
+
+  describe('step success', async () => {
+    it('we complete the first tutorial', async () => {
+      expect(screen.getByText(/Lovely\. You have completed the tutorial/)).toBeInTheDocument()
+    })
+
+    it('we can exit tutorial mode', async () => {
+      await user.click(screen.getByText('Play on my own'))
+      await act(vi.runOnlyPendingTimersAsync)
+      expect(() => screen.getByTestId('tutorial-step')).toThrow('Unable to find an element')
+    })
+
+    it('we can continue to advanced tutorial', async () => {
+      expect(screen.getByText('Learn more')).toBeInTheDocument()
+    })
```

Comment:
There's a contradiction here: If the user just exited the tutorial, then how could the "Learn more" button be visible?

This seems to rely on the tutorial inadvertently being re-launched on `beforeEach`, which does not reflect real-world usage. We should make this more explicit, as it's not super obvious what's happening here, and closer to end-user behavior. I recommend one of the following options:

1. Don't click "Play on my own" at all. Then the test can click "Learn more" just as the user would. (less test coverage)
2. Click "Play on my own". Then navigate to the Help and start the second Tutorial as the user would. (more work, but more test coverage)

Reply:
since I can do it, given the circumstances, I left it as is
at least it tests something that can break easily
and if we'll change our setup to have correct dom each time, this case will fail right away